File: ConsorcioGestFront/ConsorcioGestWeb/src/app/Models/DTO/FilterClaimDTO.ts
import { Utils } from "src/app/Helpers/Utils";

export class BaseFilterClaimDTO {
  causeClaim: number = 0;
  nroReclamo?: string;
  dateFrom?: string;
  dateTo: string = Utils.parseDate(new Date());
}

export class FilterClaimDTO extends BaseFilterClaimDTO {
  stateID: number;

  constructor() {
      super();
      this.stateID = 0;
  }
}

export class FilterClaimUserDTO extends BaseFilterClaimDTO {

}

File: ConsorcioGestBack/BusinessService/DTO/FIlterClaimDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessService.DTO
{
    public class BaseFilterClaimDTO
    {
        public int CauseClaim { get; set; }
        public string? NroReclamo { get; set; }
        public DateTime? DateFrom { get; set; }
        public DateTime? DateTo { get; set; } = DateTime.Now.Date;
    }

    public class FilterClaimDTO : BaseFilterClaimDTO
    {
        public int StateID { get; set; }
    }

    public class FilterClaimUserDTO : BaseFilterClaimDTO
    {
        
    }
}
